### DISCUSSION TITLE
Urgent: Change Classroom-generated assignment repos to private

### DISCUSSION THREAD

**Discussion by lindsaymarkward** [community/community#132182] [GitHub Education] [answered]
### Hi everyone,

Some of my student assignment repos are public, and we cannot change them to private because they are forks. 
_"For security reasons, you cannot change the visibility of a fork."_
This is absolutely not what we want - assignment work in this subject **must stay private**.
I believe that the ones generated before 1st of July are OK (private), but those after GitHub's change are public.
Is there any way to force this?
What options do I have for hiding these public assignments from view without deleting them?
They have not been marked yet, so I need access to the commit history as well as the work.
Thank you.

**Comment by catunderrain** [comment]
Hi! I'm sorry to hear about your issue, that's quite unfortunate. By the way, you can create a **new private repo**, then **copy** (mirror) the **old data** from the **old repo** to the **new private one**. For example:
```
git clone --bare https://github.com/username/old-repo.git
cd old-repo.git
git push --mirror https://github.com/username/new-private-repo.git
cd ..
rm -rf old-repo.git
```
After that, you can delete the old repo's data or change it to read-only to prevent people from accessing it. If the amount of work isn't too large, you could also consider about manually copying the files, although it might take a bit more time. The last thing is contact to [Github Support](https://support.github.com/) and may them give you some special solutions. Good luck my friend!

**Comment by lindsaymarkward** [comment, marked as answer]
Thanks so much for the response.
I started doing this and then noticed one that I was deleting COULD be changed to public. 
I believe this is because I had deleted the repo it was forked from, so it was no longer considered a fork itself.
I could then find the repo that others said they were forked from and change it to private!
This then updated which repo the others said they were forked from (who knows why!?) and I could proceed this way until all of them were done and made private.

I really don't understand what was happening, but I have managed to make my public repos private now.

**Comment by jennirinker** [comment]
I haven't tested this, but another option might be to break the fork relationship, which should allow you to change the visibility individually. But you would need to manually break the fork on every repo. Here is an article on how to detach a fork: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/detaching-a-fork
